nemo/collections/nlp/data/datasets/sgd_dataset/input_example.py

```diff
@@ -331,19 +355,25 @@ def add_noncategorical_slots(self, state_update, system_span_boundaries, user_sp
             elif values[0] == STR_DONTCARE:
                 self.noncategorical_slot_status[slot_idx] = STATUS_DONTCARE
             else:
-                self.noncategorical_slot_status[slot_idx] = STATUS_ACTIVE
                 # Add indices of the start and end tokens for the first encountered
                 # value. Spans in user utterance are prioritized over the system
                 # utterance. If a span is not found, the slot value is ignored.
                 if slot in user_span_boundaries:
+                    self.noncategorical_slot_status[slot_idx] = STATUS_ACTIVE
                     start, end = user_span_boundaries[slot]
                 elif slot in system_span_boundaries:
+                    self.noncategorical_slot_status[slot_idx] = STATUS_ACTIVE
                     start, end = system_span_boundaries[slot]
                 else:
                     # A span may not be found because the value was cropped out or because
                     # the value was mentioned earlier in the dialogue. Since this model
                     # only makes use of the last two utterances to predict state updates,
                     # it will fail in such cases.
+                    if self._add_carry_status:
+                        self.noncategorical_slot_status[slot_idx] = STATUS_CARRY
+                    else:
+                        self.noncategorical_slot_status[slot_idx] = STATUS_ACTIVE
+
```

Comment:
Please move `self.noncategorical_slot_status[slot_idx] = STATUS_ACTIVE` back directly after else, the change happens only when self._add_carry_status is True

Reply:
done